src/phoenix/experimental/callbacks/langchain_tracer.py

```diff
@@ -68,10 +76,50 @@ def _prompt_template(serialized: Dict[str, Any]) -> Iterator[Tuple[str, Any]]:
                 yield LLM_PROMPT_TEMPLATE_VARIABLES, kwargs.get("input_variables", [])
                 yield LLM_PROMPT_TEMPLATE_VERSION, "unknown"
                 break
-        except TypeError:
+        except (AttributeError, KeyError, TypeError):
             continue
 
 
+def _invocation_parameters(extra: Dict[str, Any]) -> Iterator[Tuple[str, str]]:
+    """Yields invocation parameters if present."""
+    yield LLM_INVOCATION_PARAMETERS, json.dumps(extra.get("invocation_params", {}))
+
+
+def _model_name(extra: Dict[str, Any]) -> Iterator[Tuple[str, str]]:
+    """Yields model name if present."""
+    for key in ["model_name", "model"]:
+        try:
+            yield LLM_MODEL_NAME, extra["invocation_params"][key]
```

Comment:
this makes the model name free-form. Which might be okay to start - but we are going to need to have some sort of lookup and maintain model identifiers - otherwise we might get varying names depending on framework. Thoughts? Is the idea that this key will just be the friendly name?

Reply:
Right now, I think it is just a friendly name. The model name that appears in the invocation parameters is just the argument that is passed to the `model_name` or `model` parameter. I expect it's possible for the user to provide an invalid model name, and LangChain probably just forwards those model names to the foundation model API.

Not sure I understand the need for maintaining model identifiers beyond the model names? The names are already pretty unique.
- gpt-3.5-turbo
- gpt-4
- claude-2

Reply:
Yea, model names are all made-up. Free form works well here.